api/v1alpha3/conversion.go

```diff
@@ -17,20 +17,41 @@ limitations under the License.
 package v1alpha3
 
 import (
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
 	"sigs.k8s.io/cluster-api/api/v1alpha4"
+	"sigs.k8s.io/cluster-api/util/conditions"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
 func (src *Cluster) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*v1alpha4.Cluster)
 
-	return  Convert_v1alpha3_Cluster_To_v1alpha4_Cluster(src, dst, nil)
+	if err := Convert_v1alpha3_Cluster_To_v1alpha4_Cluster(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Given this is a bool and there is no timestamp associated with it, when this condition is set, its timestamp
+	// will be "now". See https://github.com/kubernetes-sigs/cluster-api/issues/3798#issuecomment-708619826 for more
+	// discussion.
+	if src.Status.ControlPlaneInitialized {
+		conditions.MarkTrue(dst, v1alpha4.ControlPlaneInitializedCondition)
+	}
```

Comment:
Doesn't this mean that every time during a conversion we'd change the timestamp of the condition and cause the api server to fire off another patch event?

Reply:
We should probably do this after we recover conditions and only set it if needed maybe?

Reply:
Doesn't `MarkTrue` & the underlying conditions helper code only change the timestamp if the status changes?

Reply:
Ah, that's right
![](https://media4.giphy.com/media/10RgsuetO4uDkY/200.gif)